test/extensions/resource_monitors/downstream_connections/config_test.cc

```diff
@@ -0,0 +1,85 @@
+#include "envoy/extensions/resource_monitors/downstream_connections/v3/downstream_connections.pb.h"
+#include "envoy/extensions/resource_monitors/downstream_connections/v3/downstream_connections.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/extensions/resource_monitors/downstream_connections/config.h"
+#include "source/server/resource_monitor_config_impl.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/options.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace DownstreamConnections {
+namespace {
+
+TEST(ActiveDownstreamConnectionsMonitorFactoryTest, CreateMonitorInvalidConfig) {
+  auto factory =
+      Registry::FactoryRegistry<Server::Configuration::ProactiveResourceMonitorFactory>::getFactory(
+          "envoy.resource_monitors.downstream_connections");
+  EXPECT_NE(factory, nullptr);
+  EXPECT_EQ("envoy.resource_monitors.downstream_connections", factory->name());
+
+  envoy::extensions::resource_monitors::downstream_connections::v3::DownstreamConnectionsConfig
+      config;
+  config.set_max_active_downstream_connections(std::numeric_limits<uint64_t>::max());
```

Comment:
Perhaps using -1 (or some other more direct invalid value), could be more clear than relying on uint64_t::max() bits to be interpreted as an int64_t value of -1.

Reply:
agree